src/kuberesources.virtualfs.ts

```diff
@@ -0,0 +1,77 @@
+import { Uri, FileSystemProvider, FileType, FileStat, FileChangeEvent, Event, EventEmitter, Disposable } from 'vscode';
+import * as path from 'path';
+import * as fs from 'fs';
+import * as querystring from 'querystring';
+
+import { Kubectl } from './kubectl';
+import { Host } from './host';
+
+export const K8S_RESOURCE_SCHEME = "k8smsx";
```

Comment:
Kubernetes Microsoft (VS Code) extension.  I didn't want to use `k8s://` because that felt like there was a significant risk of collisions.

For my next trick: GUIDs as URI schemes!